tests/test_models.py

```diff
@@ -502,6 +510,49 @@ def _check_constraints(self):
         entity_norms = self.model.entity_embeddings.weight.norm(p=2, dim=-1)
         assert torch.allclose(entity_norms, torch.ones_like(entity_norms))
 
+    def _test_score_all_triples(self, k: Optional[int], batch_size: int = 16):
+        """Test score_all_triples.
+
+        :param k: The number of triples to return. Set to None, to keep all.
+        :param batch_size: The batch size to use for calculating scores.
+        """
+        top_triples = self.model.score_all_triples(k=k, batch_size=batch_size)
+
+        # check type
+        assert torch.is_tensor(top_triples)
+        assert top_triples.dtype == torch.long
+
+        # check shape
+        actual_k, n_cols = top_triples.shape
+        assert n_cols == 3
+        if k is None:
+            assert actual_k == self.factory.num_entities ** 2 * self.factory.num_relations
+        else:
+            assert actual_k == min(k, self.factory.num_triples)
+
+        # check ID ranges
+        assert (top_triples >= 0).all()
+        assert top_triples[:, [0, 2]].max() < self.model.num_entities
+        assert top_triples[:, 1].max() < self.model.num_relations
+
+    def test_score_all_triples(self):
+        """Test score_all_triples with a large batch size."""
+        # this is only done in one of the models
+        self._test_score_all_triples(k=15, batch_size=16)
+
+    def test_score_all_triples_singleton_batch(self):
```

Comment:
@mberr I added this test, because using batch size of 1 fails

Reply:
should be fixed by 5d6d2d4